Sources/Linq2DynamoDb.DataContext/DataContext.cs

```diff
@@ -269,20 +275,28 @@ private Table GetTableDefinition(Type entityType)
                 throw new InvalidOperationException("An instance of AmazonDynamoDbClient was not provided. Use either a ctor, that takes AmazonDynamoDbClient instance or GetTable() method, that takes Table object");
             }
 
-            var cachedTableDefinitions = _cachedTableDefinitions;
             if
             (
-                (cachedTableDefinitions == null)
+                (_cachedTableDefinitions == null)
                 ||
-                (!cachedTableDefinitions.IsAssignedToThisClientInstance(this._client))
+                (!_cachedTableDefinitions.IsAssignedToThisClientInstance(this._client))
             )
             {
-                cachedTableDefinitions = new CachedTableDefinitions(this._client);
-                _cachedTableDefinitions = cachedTableDefinitions;
```

Comment:
This is where it is replacing the shared static instance, used by all lightweight instances of all data context's.